### DISCUSSION TITLE
Help with App Registrations

### DISCUSSION THREAD

**Discussion by tonyhogsten** [Azure/ipam#210] [Q&A]
Hey!

I'm trying to deploy this, so far just running it locally with docker compose.

The problem when I start up the app, is that I am getting this type of consent window:

![requires-admin-approval](https://github.com/Azure/ipam/assets/20438742/3c84174a-5e82-4ce4-aa44-af6b2fab962a)

I have granted all the permissions as per the deploy.ps1 -AppsOnly, but still I get this window. Does anyone have any ideas as to why? And how I can get further in the troubleshooting?

**Comment by DCMattyG** [comment]
Hi there @tonyhogsten, so happy to assist you with this issue you are facing!

It would seem that whomever executed the deployment script with the `-AppsOnly` flag was not a [Global Admin](https://learn.microsoft.com/en-us/entra/identity/role-based-access-control/permissions-reference#global-administrator). This is a requirement that is mentioned [here](https://azure.github.io/ipam/#/deployment/README?id=prerequisites) in our prerequisites section. The reason this is important is because Admin consent needs to be granted for the App Registrations so that users aren't prompted repeatedly to agree to consent. Additionally, because there is a frontend and backend component, users can't be presented with consent for the backend application so many operations will fail.

Example of granted consent...

![image](https://github.com/Azure/ipam/assets/12955732/d6a94c1e-4cac-4052-a6b7-5b5866d886ce)

![image](https://github.com/Azure/ipam/assets/12955732/ea426152-71ae-4c97-bbde-6530a2ff348b)

So it's likely that you'll want to either redeploy the solution with the proper permissions so that the deployment script can set everything on your behalf, or manually correct the missing permissions and/or consent for the App Registrations and possible the Reader permissions for the Engine Service Principal to the Root Management Group.

I certainly hope this has been informative and helpful, but if you need anything else please don't hesitate to reach out. I'm happy to suggest additional fixes or even jump into a 1:1 Teams call with you for troubleshooting. Just let us know!

**Comment by tonyhogsten** [comment]
Hey @DCMattyG!

Thanks for you answer. I managed to get this resolved. It was granted according to your guide, but we added a permission that we forgot (We did this manually, as no one has Global Admin and it's impossible to get someone with those rights to run the script). When adding this right, it probably broke the granting (Although the interface in Entra ID was still showing `Granted for X..`). 

The people who can grant delegations in my organisation re-granted and then it worked. 

Thanks for your help!
